store/views.py

```diff
@@ -77,6 +97,44 @@ def loanBookView(request):
 @csrf_exempt
 @login_required
 def returnBookView(request):
-    pass
+    response_data={
+        'message':None,
+    }
+    book_id=request.POST['bid']
+    book=BookCopy.objects.get(pk=book_id)
+    try:
+        book.borrower=None
+        book.borrow_date=None
+        book.status=True
+        book.save()
+        msg="success"
+    except:
+        msg='failure'
+    response_data['message']=msg
+    return JsonResponse(response_data)        
+
+@csrf_exempt
+@login_required
+def rateView(request, id):
+    if request.method == "POST":
+        book = Book.objects.get(pk=id)
+        r1=UserRating.objects.filter(user=request.user,book=book)
+        user_rating = request.POST['user_rating']
+        rating=UserRating()
+        rating.book=book
+        rating.user=request.user
+        rating.rating=user_rating
+        r1.delete()
+        rating.save()
```

Comment:
You've not put a backend validation on the rating, so the user can easily put invalid values of rating.

Also, you could have updated the rating rather than deleting and then saving it.